emperor/core.py

```diff
@@ -377,29 +401,77 @@ def make_emperor(self, standalone=False, custom_axes=None,
 
         main_template = self._get_template(standalone)
 
-        coord_ids, coords, pct_var, ci, headers, metadata, names = \
-            self._process_data(custom_axes, jackknifing_method)
+        # _process_data does a lot of munging to the coordinates data and
+        # _to_dict puts the data into a dictionary-like object for consumption
+        data = self._to_dict(self._process_data(custom_axes,
+                                                jackknifing_method))
 
         # yes, we could have used UUID, but we couldn't find an easier way to
         # test that deterministically and with this approach we can seed the
         # random number generator and test accordingly
         plot_id = 'emperor-notebook-' + str(hex(np.random.randint(2**32)))
 
         # need to do something about low and high
-        plot = main_template.render(coords_ids=coord_ids, coords=coords,
-                                    pct_var=pct_var, ci=ci,
-                                    md_headers=headers, metadata=metadata,
-                                    plot_id=plot_id,
-                                    axes_names=names,
-                                    base_url=self.base_url,
+        plot = main_template.render(data=data, plot_id=plot_id,
                                     logic_template_path=basename(LOGIC_PATH),
                                     style_template_path=basename(STYLE_PATH),
+                                    base_url=self.base_url,
                                     width=self.width,
-                                    height=self.height,
-                                    settings=self.settings)
+                                    height=self.height)
 
         return plot
 
+    def _to_dict(self, data):
+        """Convert processed data into a dictionary of decompositions
+
+        Parameters
+        ----------
+        data : tuple
+            The output of _process_data. Should contain information about the
+            scatter plot and the biplot.
+
+        Returns
+        -------
+        dict
+            A dictionary describing the plots contained in the ordination
+            object and the sample + feature metadata.
+        """
+        # data is a tuple as returned by _process_data
```

Comment:
Would it make sense for `_process_data` to return a `namedtuple`?

Reply:
I don't think so, `_process_data` is a private method not intended for use by anything else but `_to_json`.